# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1236443731 18000
#      Sat Mar 07 11:35:31 2009 -0500
# Node ID d47290508a55b42365205ce808110860e6396acf
# Parent  25b85c36208ff7a6523f35bf70f015a68d3835c2
document --doc-cache-file option

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,13 @@
+2009-03-07  John W. Eaton  <jwe@octave.org>
+
+	* interpreter/basics.txi (Command Line Options):
+	Document --doc-cache-file command-line option.
+
 2009-03-06  Jaroslav Hajek  <highegg@gmail.com>
 
 	* interpreter/diagperm.txi: Various improvements.
 
 2009-02-27  John W. Eaton  <jwe@octave.org>
 
 	* interpreter/set.txi (Sets): Don't document create_set.
 
diff --git a/doc/interpreter/basics.txi b/doc/interpreter/basics.txi
--- a/doc/interpreter/basics.txi
+++ b/doc/interpreter/basics.txi
@@ -71,16 +71,24 @@ accepts.
 @item --debug
 @itemx -d
 @cindex @code{--debug}
 @cindex @code{-d}
 Enter parser debugging mode.  Using this option will cause Octave's
 parser to print a lot of information about the commands it reads, and is
 probably only useful if you are actually trying to debug the parser.
 
+@item --doc-cache-file
+@cindex @code{--doc-cache-file} @var{filename}
+Specify the name of the doc cache file to use.  The value of @var{filename}
+specified on the command line will override any value of
+@code{OCTAVE_DOC_CACHE_FILE} found in the environment, but not any commands
+in the system or user startup files that use the @code{doc_file_cache}
+function.
+
 @item --echo-commands
 @itemx -x
 @cindex @code{--echo-commands}
 @cindex @code{-x}
 Echo commands as they are executed.
 
 @item --eval @var{code}
 Evaluate @var{code} and exit when done unless @code{--persist} is also
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2009-03-07  John W. Eaton  <jwe@octave.org>
+
+	* octave.cc (verbose_usage): Include --doc-cache-file in option list.
+
 2009-03-06  Jaroslav Hajek  <highegg@gmail.com>
 
 	* ov.h (octave_value::diag_matrix_value,
 	octave_value::complex_diag_matrix_value,
 	octave_value::float_diag_matrix_value,
 	octave_value::float_complex_diag_matrix_value,
 	octave_value::perm_matrix_value): New methods.
 
diff --git a/src/octave.cc b/src/octave.cc
--- a/src/octave.cc
+++ b/src/octave.cc
@@ -482,16 +482,17 @@ verbose_usage (void)
 {
   std::cout << OCTAVE_NAME_VERSION_COPYRIGHT_COPYING_AND_WARRANTY "\n\
 \n\
 Usage: octave [options]\n\
 \n\
 Options:\n\
 \n\
   --debug, -d             Enter parser debugging mode.\n\
+  --doc-cache-file FILE   Use doc cache file FILE.\n\
   --echo-commands, -x     Echo commands as they are executed.\n\
   --eval CODE             Evaluate CODE.  Exit when done unless --persist.\n\
   --exec-path PATH        Set path for executing subprograms.\n\
   --help, -h, -?          Print short help message and exit.\n\
   --image-path PATH       Set initial IMAGE_PATH to PATH.\n\
   --info-file FILE        Use top-level info file FILE.\n\
   --info-program PROGRAM  Use PROGRAM for reading info files.\n\
   --interactive, -i       Force interactive behavior.\n\
